time_manager/time_manager.F90

```diff
@@ -2140,24 +2162,65 @@ function set_date_gregorian(year, month, day, hour, minute, second, tick, Time_o
  type(time_type),  intent(out) :: Time_out
  character(len=*), intent(out) :: err_msg
  integer :: yr1, day1
+ integer :: ncenturies, nlpyrs, l
 
  if( .not.valid_increments(year,month,day,hour,minute,second,tick,err_msg) ) then
    set_date_gregorian = .false.
    return
  endif
 
+ l = 0 ; if( leap_year_gregorian_int(year) ) l = 1
+
+ ! Check if date is invalid
+ if( month.ne.2 ) then
+    if ( day.gt.days_per_month(month) .or. day.lt.1 ) then
+       err_msg = 'Invalid_date. Date='//convert_integer_date_to_char(year,month,day,hour,minute,second)
+       set_date_gregorian = .false.
+       return
+    end if
+ else if (month.eq.2 ) then
+    if ( day.gt.days_per_month(month)+l .or. day.lt.1 ) then
+       err_msg = 'Invalid_date. Date='//convert_integer_date_to_char(year,month,day,hour,minute,second)
+       set_date_gregorian = .false.
+       return
+    end if
+ end if
+
  Time_out%seconds = second + 60*(minute + 60*hour)
 
- yr1 = mod(year,400)
- if(yr1 == 0) yr1 = 400
- day1 = date_to_day(yr1,month,day)
-  if(day1 == invalid_date) then
+ yr1  = mod(year-1,400)
+ day1 = 0
+ if( yr1.gt.0 ) then
```

Comment:
Do we not support negative years in the Gregorian calendar?

Reply:
@mlee03 - was this comment from @underwoo regarding negative year support resolved?  How does the old method handle it?